ecommerce-ui/src/App.js

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import './App.css';
+import ShoppingCart from './ShoppingCart';
+import HouseListings from './HouseListings';
+import _listingData from './airbnbs.json';
+
+class App extends React.Component {
+  state = {
+    houses: _listingData,
+    selectedHouses: [],
+    totalCost: 0
+  }
+
+  addHouse = (idx) => {
+    const selectedHouse = this.state.houses[idx];
+    const selectedCost = this.state.houses[idx].payment.cost;
+    
+    if (this.state.selectedHouses.includes(selectedHouse)){
+      return;
+    }
+
+    this.setState(prevState => {
+      return {
+        totalCost: (this.state.totalCost + selectedCost),
+        selectedHouses: [...prevState.selectedHouses, selectedHouse]
+      };
+    });
+  }
+  
+  removeHouse = (idx) => {
+    const selectedCost = this.state.houses[idx].payment.cost;
```

Comment:
The cart total doesn't update correctly when going back and forth between adding/removing houses. It looks like that's due to using an index for the selected house array to access a house in the houses array, which is causing the wrong costs to be subtracted. This could be fixed by either 1. Getting the selected cost from the selectedHouses array, or 2. Not storing the total cost in state and computing it in the render function.